typings/index.d.ts

```diff
@@ -431,6 +431,56 @@ export class CommandInteraction extends Interaction {
   private _createOptionsCollection(options: unknown, resolved: unknown): Collection<string, CommandInteractionOption>;
 }
 
+export class CommandInteractionOptionResolver {
+  constructor(options: CommandInteractionOption[]);
+  private options: CommandInteractionOption[];
+  private _getTypedOption(
+    name: string,
+    types: ApplicationCommandOptionType[],
+    properties: (keyof ApplicationCommandOption)[],
+    required: true,
+  ): CommandInteractionOption;
+  private _getTypedOption(
+    name: string,
+    types: ApplicationCommandOptionType[],
+    properties: (keyof ApplicationCommandOption)[],
+    required: boolean,
+  ): CommandInteractionOption | null;
+
+  public get(name: string, required: true): CommandInteractionOption;
+  public get(name: string, required?: boolean): CommandInteractionOption | null;
+  public getSubCommand(name: string): CommandInteractionOptionResolver | null;
+  public getBoolean(name: string, required: true): boolean;
+  public getBoolean(name: string, required?: boolean): boolean | null;
+  public getChannel(name: string, required: true): NonNullable<CommandInteractionOption['channel']>;
+  public getChannel(name: string, required?: boolean): NonNullable<CommandInteractionOption['channel']> | null;
+  public getString(name: string, required: true): string;
+  public getString(name: string, required?: boolean): string | null;
+  public getInteger(name: string, required: true): number;
+  public getInteger(name: string, required?: boolean): number | null;
+  public getUser(name: string, required: true): NonNullable<CommandInteractionOption['user']>;
+  public getUser(name: string, required?: boolean): NonNullable<CommandInteractionOption['user']> | null;
+  public getMember(name: string, required: true): NonNullable<CommandInteractionOption['member']>;
+  public getMember(name: string, required?: boolean): NonNullable<CommandInteractionOption['member']> | null;
+  public getRole(name: string, required: true): NonNullable<CommandInteractionOption['role']>;
+  public getRole(name: string, required?: boolean): NonNullable<CommandInteractionOption['role']> | null;
+  public getMentionable(
+    name: string,
+    required: true,
+  ):
+    | NonNullable<CommandInteractionOption['member']>
+    | NonNullable<CommandInteractionOption['role']>
+    | NonNullable<CommandInteractionOption['user']>;
+  public getMentionable(
+    name: string,
+    required?: boolean,
+  ):
+    | NonNullable<CommandInteractionOption['member']>
+    | NonNullable<CommandInteractionOption['role']>
+    | NonNullable<CommandInteractionOption['user']>
+    | null;
```

Comment:
You can simplify the return types:
```suggestion
  public getMentionable(
    name: string,
    required: true,
  ): NonNullable<CommandInteractionOption['member' | 'user' | 'role']>;
  public getMentionable(
    name: string,
    required?: boolean,
  ): NonNullable<CommandInteractionOption['member' | 'user' | 'role']> | null;
```

Reply:
TIL